compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/LocalDeclarationsLowering.kt

```diff
@@ -953,7 +974,15 @@ class LocalDeclarationsLowering(
                     // LDL doesn't work on the enums because local enums are not allowed, so skipping them in scripts too
                     if (isScriptMode && declaration.kind == ClassKind.ENUM_CLASS) return
 
-                    localClasses[declaration] = LocalClassContext(declaration, data.inInlineFunctionScope)
+                    // If there are many primary constructors, each copy of the initializer requires different remapping.
```

Comment:
I'm out of the loop... How is it possible to have many primary constructors in the same class? O_o

Reply:
An even more naive question: why not use `IrConstructor.isPrimary`?

Reply:
Not sure if calling them "primary" is correct, but:
```
class C {
    constructor() {}
    constructor(x: Int) {}
    val y = something // copied into both constructors
}
```
This is why InitializersLowering does deep-copying, and why JvmLocalClassPopupLowering is a thing at all.

The `callsSuper` stuff is borrowed from [another part of LocalDeclarationsLowering](https://github.com/JetBrains/kotlin/blob/0864f9f/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/LocalDeclarationsLowering.kt#L473-L477). `isPrimary` (or `primaryConstructor`) didn't work for some reason for some edge case, but I don't remember what it was, I'll rerun the tests tomorrow. Maybe instead of `callsSuper` both places should check for an `IrInstanceInitializerCall` though.

Reply:
`localClassConstructors[declaration.primaryConstructor]` *does* seem to work so I don't know what I did wrong previously, but obviously it doesn't optimize the above code even if it only has one constructor.

As for `IrInstanceInitializerCall`, looks like sometimes we forget to add it in other lowerings...

Reply:
Ok, "sometimes" means only in SuspendLambdaLowering.